src/screens/details.tsx

```diff
@@ -1,19 +1,209 @@
-import { StyleSheet, View } from 'react-native'
-import { Text } from 'react-native-paper'
+import { useState } from 'react'
+import { FlatList, StyleSheet, View, Text } from 'react-native'
+import { Button } from 'react-native-paper'
+import DropDown from 'react-native-paper-dropdown'
+import { config } from '../config'
 
-const DetailsScreen = () => (
-  <View style={styles.container}>
-    <Text>Details Screen</Text>
-  </View>
-)
+const tfgmEndpoint = 'https://api.tfgm.com/odata/Metrolinks'
 
-export default DetailsScreen
+type APIType = {
+  AtcoCode: string
+  StationLocation: string
+  Dest0: string
+  Wait0: string
+  Status0: string
+  Carriages0: string
+  Dest1: string
+  Wait1: string
+  Status1: string
+  Carriages1: string
+  Dest2: string
+  Wait2: string
+  Status2: string
+  Carriages2: string
+}[]
+
+type IncomingTram = {
+  dest: string
+  wait: string
+  carriages: string
+  status: string
+}
+
+class StopData {
+  stopName = ''
+  incomingTrams: IncomingTram[] = []
+}
+
+export default function DetailsScreen() {
+  const [showDropDown, setShowDropDown] = useState(false)
+  const [stop, setStop] = useState('')
+  const [incomingTrams, setIncomingTrams] = useState([] as IncomingTram[])
+  const [stopsObtained, setStopsObtained] = useState(
+    [] as { label: string; value: string }[]
+  )
+
+  async function getAllStops() {
+    if (stopsObtained.length > 0) {
+      return
+    }
+
+    const stops: { label: string; value: string }[] = []
+    const res = await fetch(tfgmEndpoint, {
+      method: 'GET',
+      headers: {
+        'Ocp-Apim-Subscription-Key': config.apiKey,
+      },
+    })
+    const json = (await res.json()) as { value: APIType }
+    for (const item of json.value) {
+      const truncAtcoCode = item.AtcoCode.substring(0, item.AtcoCode.length - 1)
+      const newStop = { label: item.StationLocation, value: truncAtcoCode }
+
+      if (
+        !stops.some((value) => {
+          return value.value === truncAtcoCode
+        })
```

Comment:
You can get rid of the curly brackets here.

Very minor thing but I think using `every` and `!==` here makes it easier to read. I'm not sure, do you agree?

Reply:
I read my code as "If there is not an atco code in `stops` which has the same atco code as the new stop, push the new stop"

I read the `every`/`!==` as "If all atco codes in `stops` do not have the same atco code as the new stop, push the new stop" which, at least to me, doesn't fully explain what's being checked in the context of the rest of the code

Reply:
> read the every/!== as "If all atco codes in stops do not have the same atco code as the new stop, push the new stop" which, at least to me, doesn't fully explain what's being checked in the context of the rest of the code

I see what you mean, but I could also read it as "If the new atco code isn't present in the `stops`, then add the new stop" which _is_ what you're doing.

I think either way works, thought it would be interesting to get your thoughts on it. No need to change this.